nssp/tests/test_patch.py

```diff
@@ -0,0 +1,130 @@
+import unittest
+from unittest.mock import patch as mock_patch, call
+from delphi_nssp.patch import patch, good_patch_config
+import os
+import shutil
+from datetime import datetime, timedelta
+
+class TestPatchModule(unittest.TestCase):
```

Comment:
Any particular reason for using unittest? Personally I've been using pytest and I think one of the older tickets somewhere said wanted to standardize into using pytest; more pythonic compated to unittests, which is based on Junit and such

Reply:
I'm a fan of `mock`, hence the use of `unittest`. `mock` is in `unittest`, which is in the Python standard library in recent versions of python. It's the [same mock on pypi](https://docs.python.org/3/library/unittest.mock.html):

> Mock is designed for use with [unittest](https://docs.python.org/3/library/unittest.html#module-unittest) and is based on the ‘action -> assertion’ pattern instead of ‘record -> replay’ used by many mocking frameworks.
> 
> There is a backport of [unittest.mock](https://docs.python.org/3/library/unittest.mock.html#module-unittest.mock) for earlier versions of Python, available as [mock](https://pypi.org/project/mock/) on PyPI.

But ye lemme rewrite the rest to use pytest.